server/src/main/java/io/druid/server/DruidNode.java

```diff
@@ -128,6 +128,11 @@ public int getPort()
     return port;
   }
 
+  public boolean isListening()
+  {
+    return port >= 0;
```

Comment:
in java URI world -1 signifies undefined or default. As such it is not always invalid in java to have a port of -1.